gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -493,6 +489,10 @@ public void forWorkUnit(WorkUnit workUnit) {
             }
           });
 
+          // If it is a streaming source, workunits cannot be counted
```

Comment:
Is there a unit test we can add for this? It seems like the previous location of this block was incorrect. To prevent future regression, it would be good to add a unit test for this.